controllers/issuer_controller.go

```diff
@@ -88,32 +96,37 @@ func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 				Force:        ptr.To(true),
 			},
 		}); err != nil {
-			if err := client.IgnoreNotFound(err); err != nil {
+			if !apierrors.IsNotFound(err) {
 				return ctrl.Result{}, utilerrors.NewAggregate([]error{err, returnedError})
 			}
+
 			logger.V(1).Info("Not found. Ignoring.")
 		}
 	}
 
 	return result, returnedError
 }
 
+// reconcileStatusPatch is responsible for reconciling the issuer. It will return the
+// result and reconcileError to be returned by the Reconcile function. It also returns
+// an issuerStatusPatch that the Reconcile function will apply to the issuer's status.
+// This function is split out from the Reconcile function to allow for easier testing.
```

Comment:
**Suggestion:** I think it is still unclear why the error should not be checked by the Reconcile func:

> It will return the result and reconcileError to be returned by the Reconcile function.

This doesn't explain why.

I suggest adding an extra bit that details why:

```suggestion
// reconcileStatusPatch is responsible for reconciling the issuer. It will return the
// result and reconcileError to be returned by the Reconcile function. It also returns
// an issuerStatusPatch that the Reconcile function will apply to the issuer's status.
// This function is split out from the Reconcile function to allow for easier testing.
//
// The error returned by `reconcileStatusPatch` is meant for controller-runtime,
// not for the caller. The caller must not check the error (i.e., they must not
// do `if err != nil...`).
```